test/core/end2end/generate_tests.bzl

```diff
@@ -431,35 +431,25 @@ def grpc_end2end_tests():
         )
 
         for t, topt in END2END_TESTS.items():
-            #print(_compatible(fopt, topt), f, t, fopt, topt)
             if not _compatible(fopt, topt):
                 continue
-
             test_short_name = str(t) if not topt.short_name else topt.short_name
-            native.sh_test(
+            grpc_cc_test(
                 name = "%s_test@%s" % (f, test_short_name),
-                data = [":%s_test" % f],
-                srcs = ["end2end_test.sh"],
-                args = [
-                    "$(location %s_test)" % f,
-                    t,
+                srcs = ["fixtures/%s.cc" % f],
+                data = [":end2end_tests"],
```

Comment:
Seems odd to have the same target listed in both deps and data.  Why is this needed?  If it's related to the TSI test certificate files above, maybe just move that data here instead?

Reply:
As per my preference above, we'd be duplicating the TSI cert data to two targets if we did that. If (in a separate PR) we can remove the end2end test cc_binary target, then you're exactly right, all we'd need to do is move the data here.